semeio/workflows/misfit_preprocessor/job.py

```diff
@@ -4,30 +4,23 @@
 
 def run(config, measured_data, reporter):
     workflow = config.workflow
-    if workflow.type == "custom_scale":
-        sconfig = workflow.clustering.cluster_args()
-        scaling_configs = spearman_job(
-            measured_data,
-            reporter,
-            **sconfig,
-        )
-        pca_threshold = config.workflow.pca.threshold
-    elif workflow.type == "auto_scale":
+    sconfig = workflow.clustering.cluster_args()
+    if workflow.type == "auto_scale":
         job = ObservationScaleFactor(reporter, measured_data)
         nr_components, _ = job.perform_pca(workflow.pca.threshold)
-        sconfig = workflow.clustering.cluster_args()
-        scaling_configs = spearman_job(
-            measured_data,
-            reporter,
-            criterion="maxclust",
-            threshold=nr_components,
-            **sconfig,
-        )
-        pca_threshold = workflow.pca.threshold
-    else:
-        raise AssertionError(
-            "Unknown clustering method: {}".format(config.workflow.method)
-        )
+        if workflow.clustering.type == "limited_hierarchical":
+            sconfig["criterion"] = "maxclust"
```

Comment:
I am still against modifying the configuration, even if it is set to `None` before we get to this place.

I would want the `sconfig` object to be immutable. A possible implementation would then be to add a `job_kwargs = {}`, update that instead of the `sconfig` and pass it to the function along with `sconfig`: `spearman_job(**sconfig, **job_kwargs)`

Reply:
But we are not modifying the configuration though, we are adding things that are not present in the configuration.

Reply:
even if ~it is set to None~ it does not exist before we get to this place.

Reply:
But is that kind of not what we are doing? It is just a dict (`sconfig`), a copy of the properties of the scaling function, the actual configuration object (`workflow`) is not being modified in any way.

Reply:
I didn't look properly where the `sconfig` came from (....`workflow.cluster.cluster_args()`). I thought we were changing the instance coming from pydantic. I agree that passing just the dicts makes for a better interface.